### PR TITLE
[v2] update to SBO library to 1.0

### PR DESCRIPTION
**Pull request by feloy**
Fixes https://github.com/redhat-developer/odo/issues/5221

- [ ] Binding Redis instance fails if instance created without redisSecret field (https://github.com/redhat-developer/service-binding-operator/issues/1088)
- [x] unlinking two components with SBO installed fails: ListOperatorServices was breaking after the firdt empty list found
- [x] no SBO installed and non-admin user and link to component: error no binding data to project

If you delete the Service Binding Operator from your OpenShift cluster, IsResourceSupported continues to detect it => in this circumstance, odo will create sbr for binding, and will timeout waiting for deployment to be restarted after the sbr is created. SOLUTION: delete servicebindings CRDs manually

### CODE REVIEW

**Comment by kadel** [comment]
> [@kadel](/kadel) [kadel](/kadel) closed this [15 seconds ago](#event-6050641838)
> [@kadel](/kadel) [kadel](/kadel) reopened this [11 seconds ago](#event-6050642241)

:innocent: wrong button, sorry :blush:

**Comment by feloy** [comment]
> Talked to SBO team about [redhat-developer/service-binding-operator#1088](https://github.com/redhat-developer/service-binding-operator/issues/1088) they won't have the bandwidth to look into [redhat-developer/service-binding-operator#1088](https://github.com/redhat-developer/service-binding-operator/issues/1088) in the next 2 sprints (probably even more). It looks like we will have to implement one of the workarounds on our side.
> 
> The easiest workaround on our side could be simply updating our tests to include creation of a `Secret` and defining `kubernetesConfig.redisSecret.name` in Redis CR

OK, this is actually what is done in this PR. I'll mark the PR as ready for review

Thanks @kadel

**Comment by kadel** [comment]
> OK, this is actually what is done in this PR. I'll mark the PR as ready for review

oh, my bad. I should have checked the code of this PR

File: tests/integration/operatorhub/cmd_service_test.go [hunk 1]
```diff
@@ -169,6 +169,8 @@ var _ = Describe("odo service command tests for OperatorHub", func() {
 							"-p", "kubernetesConfig.serviceType=ClusterIP",
 							"-p", "kubernetesConfig.resources.requests.cpu=100m",
 							"-p", "kubernetesConfig.resources.requests.memory=128Mi",
+							"-p", "kubernetesConfig.redisSecret.name=redis-secret",
```
Comment (kadel) [inline tests/integration/operatorhub/cmd_service_test.go:172]: :+1:

**Review by kadel — COMMENTED** [review]
/approve

**Pull request merged**